DevKit/DevKitTests/Classes/Data Structures/Linked List/LinkedListTests.swift

```diff
@@ -0,0 +1,155 @@
+        // Given
+        var linkedList = LinkedList<Int>()
+        linkedList.push(3)
+        linkedList.push(2)
+        linkedList.push(1)
+        XCTAssertEqual(linkedList.tail!.value, 3)
+        // When
+        linkedList.append(4)
+        // Then
+        XCTAssertEqual(linkedList.tail!.value, 4)
+    }
+
+    func test_linkedList_InsertAt() {
+        // Given
+        var linkedList = LinkedList<Int>()
+        linkedList.push(3)
+        linkedList.push(2)
+        linkedList.push(1)
+        guard let nodeAtIndex1 = linkedList.node(at: 1) else {
+            XCTFail("Node is null when it should be valid.")
+            return
+        }
+        XCTAssertEqual(nodeAtIndex1.value, 2)
+        // When
+        linkedList.insert(value: 4, after: nodeAtIndex1)
+    }
+
+    func test_linkedList_InsertAtTail() {
+        // Given
+        var linkedList = LinkedList<Int>()
+        linkedList.push(3)
+        linkedList.push(2)
+        linkedList.push(1)
+        guard let nodeAtIndex2 = linkedList.node(at: 2) else {
+            XCTFail("Node is null when it should be valid.")
+            return
+        }
+        XCTAssertEqual(nodeAtIndex2.value, linkedList.tail!.value)
+        // When
+        linkedList.insert(value: 4, after: nodeAtIndex2)
+    }
+
+    func test_linkedList_PopWhenEmpty() {
+        // Given
+        var linkedList = LinkedList<Int>()
+        // When
+        let value = linkedList.pop()
+        // Then
+        XCTAssertNil(value)
+    }
+
+    func test_linkedList_Pop() {
+        // Given
+        var linkedList = LinkedList<Int>()
+        linkedList.push(3)
+        linkedList.push(2)
+        linkedList.push(1)
+        // When
+        let value = linkedList.pop()
+        // Then
+        XCTAssertEqual(value!, 1)
+    }
+
+    func test_linkedList_RemoveLastWhenEmpty() {
+        // Given
+        var linkedList = LinkedList<Int>()
+        // When
+        let value = linkedList.removeLast()
+        // Then
+        XCTAssertNil(value)
+    }
+
+    func test_linkedList_RemoveLastWhenOnly1Element() {
+        // Given
+        var linkedList = LinkedList<Int>()
+        linkedList.push(1)
+        // When
+        let value = linkedList.removeLast()
+        // Then
+        XCTAssertEqual(value!, 1)
+    }
+
+    func test_linkedList_RemoveLast() {
+        // Given
+        var linkedList = LinkedList<Int>()
+        linkedList.push(3)
+        linkedList.push(2)
+        linkedList.push(1)
+        // When
+        let value = linkedList.removeLast()
+        // Then
+        XCTAssertEqual(value!, 3)
+    }
+
+    func test_linkedList_removeAfter() {
+        // Given
+        var linkedList = LinkedList<Int>()
+        linkedList.push(3)
+        linkedList.push(2)
+        linkedList.push(1)
+        // When
+        let value = linkedList.remove(after: linkedList.head!)
+        print("Value removed: ", value)
+        // Then
+        print(linkedList.head?.value)
```

Comment:
Do we need these print statements in the tests?

Reply:
No, removed.